intelmq/bots/collectors/file/collector_file.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""
+File Collector Bot
+
+Regularly checks a directory and reads files matching an extension from it
+It can delete the file if a delete switch was set.
+The bot creates on report per file.
+
+Copyright (C) 2016 by Bundesamt für Sicherheit in der Informationstechnik
+Software engineering by Intevation GmbH
+
+Parameters:
+path: string
+extension: string
+delete_file: boolean
+    default: True
+"""
+
+import sys
+import os
+import fnmatch
+
+from intelmq.lib.bot import Bot
+from intelmq.lib.message import Report
+
+
+class FileCollectorBot(Bot):
+
+    def init(self):
+        # Test if path can be accessed
+        try:
+            if not os.path.isdir(self.parameters.path):
+                self.logger.error("The path does not lead to " \
+                    "a directory")
+
+        except:
```

Comment:
Two issues here: first, we don't want to catch all Exceptions, only `PermissionError`
Second, `os.path.isdir` does not raise for permission errors here (for me at least). `os.stat` does.